src/molecules/tables/index.tsx

```diff
@@ -244,37 +244,45 @@ export default function DataTable<TData, TValue>(
           </TableHeader>
           <TableBody className={cn(classNames?.table?.body)}>
             {table.getRowModel().rows?.length ? (
-              table.getRowModel().rows.map((row) => (
-                <Fragment key={row.id}>
-                  <TableRow
-                    data-state={row.getIsSelected() ? 'selected' : undefined}
-                    className="whitespace-nowrap"
-                  >
-                    {row.getVisibleCells().map((cell) => {
-                      const className = cell.id.includes('actions')
-                        ? 'p-0'
-                        : 'p-2';
-                      return (
-                        <TableCell key={cell.id} className={className}>
-                          {
-                            flexRender(
-                              cell.column.columnDef.cell,
-                              cell.getContext()
-                            ) as JSX.Element
-                          }
-                        </TableCell>
-                      );
-                    })}
-                  </TableRow>
-                  {row.getIsExpanded() && renderSubComponent && (
-                    <TableRow>
-                      <TableCell colSpan={row.getVisibleCells().length}>
-                        {renderSubComponent({ row })}
-                      </TableCell>
+              table
+                .getRowModel()
+                .rows.slice(
+                  table.getState().pagination.pageIndex *
+                    table.getState().pagination.pageSize,
+                  (table.getState().pagination.pageIndex + 1) *
+                    table.getState().pagination.pageSize
```

Comment:
Let's remove the slice here and set the [maual-pagination](https://tanstack.com/table/latest/docs/api/features/pagination#manualpagination) to false in the table props, to handle the server side request we have a function called fetch request if it's there we can set it to false.

Reply:
I removed the slice function and updated manualPagination as shown below:
![image](https://github.com/user-attachments/assets/d27b198f-af87-4b3c-988a-f70cc8e604bc)

However, all rows are still displayed, ignoring the page number and rows-per-page states.
example:
![image](https://github.com/user-attachments/assets/4e98327d-730e-4db1-8b2c-8996aaa2d1d6)

I also tried setting manualPagination to false without checking for fetchRequest (since fetchRequest is undefined here), but the issue persists.

Let me know if you’d like me to push these changes regardless of this problem.

Reply:
Checking [tanstack example](https://tanstack.com/table/latest/docs/framework/react/examples/pagination) I relized that we miss a function, if you add the `getPaginationRowModel` then it should work as expected.

```diff
+    getPaginationRowModel: getPaginationRowModel(),
+    manualPagination: false,
```

Reply:
I added it and now it works.